drivers/edac/ibecc.h

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (c) 2020 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* TODO: Add to include/sys/util.h */
+#define BITFIELD(val, h, l)	(((val) & GENMASK(h, l)) >> l)
+
+#define EDAC_IBECC_NAME		"IBECC"
+
+#define PCI_VENDOR_ID_INTEL	0x8086
+#define PCI_DEVICE_ID_SKU7	0x452a
+#define PCI_DEVICE_ID_SKU12	0x4518
+
+/* TODO: Move to correct place NMI registers */
+
+/* NMI Status and Control Register (NMI_STS_CNT) */
+#define NMI_STS_CNT_REG		0x61
```

Comment:
Minor thing, but applies to everywhere where you have defines etc: please only use tabs for indentation and never for alignment (i.e. only spaces should be used for that). Here you've used tabs for alignment which makes this render inconsistently e.g. in a browser when reviewing using the GitHub web interface.

Reply:
> please take into account the comment from @nashif about magic numbers, and you should probably next look into making this properly use Device Tree

Perhaps this could be an enhancement?

Reply:
if it is not fixed in this PR, it will just get lost. Something like that is not worth opening an enhancement issue for.

Reply:
> Minor thing, but applies to everywhere where you have defines etc: please only use tabs for indentation and never for alignment (i.e. only spaces should be used for that). Here you've used tabs for alignment which makes this render inconsistently e.g. in a browser when reviewing using the GitHub web interface.

Is this mentioned in the Coding Style document? I think I usually see in headers alignments by tabs, like in include/drivers/pcie/pcie.h, etc.

Reply:
@nashif @jhedberg @jenmwms I can make driver use Device Tree, just need to invent what kind of useless information I am going to get from it ;)

Reply:
> Is this mentioned in the Coding Style document? I think I usually see in headers alignments by tabs, like in include/drivers/pcie/pcie.h, etc.

I'm not sure if it's mentioned explicitly anywhere. And yes, the Zephyr tree is wildly inconsistent with this. However, what I've found is that "tabs for indentation, spaces for alignment" is what gives the most readable result in multiple UIs (both editors and web browsers).

Reply:
I have created simple DTS: 5735ba12c0e7f38dd342cb82ba71d864de960e8e. I pass BDF and Intel Vendor ID, well known and harmful identifiers. For the product ID we might need to pass list of IDs.

Reply:
> @nashif @jhedberg @jenmwms I can make driver use Device Tree, just need to invent what kind of useless information I am going to get from it ;)

I am not referring to DT, just the one magic number and the need to define it somewhere with details about where it came from.

Reply:
@nashif I will fix this, there might be some other magic numbers like `GENMASK(31, 20)`, I might fix it also.

Reply:
Added 06c874712575025ee5f767936c3e11d27437b20e